toolchain/sem_ir/ids.cpp

```diff
@@ -59,6 +61,13 @@ auto BoolValue::Print(llvm::raw_ostream& out) const -> void {
   }
 }
 
+auto CharId::Print(llvm::raw_ostream& out) const -> void {
+  // TODO: If we switch to C++23, `std::format("`{:?}`")` might be a better
+  // choice.
+  out << "U+";
+  llvm::write_hex(out, index, llvm::HexPrintStyle::Upper);
```

Comment:
```suggestion
  llvm::write_hex(out, index, llvm::HexPrintStyle::Upper, 4);
```
[U+nnnn notation should always include at least 4 hexadecimal digits](https://en.wikipedia.org/wiki/Unicode#Codespace_and_code_points:~:text=At%20least%20four%20hexadecimal%20digits%20are%20always%20written).